UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Webjob.UnitTests/Services/FulfilmentFileShareServiceTest.cs

```diff
@@ -132,23 +137,56 @@ public async Task WhenRequestDownloadReadMeFile_ThenReturnsTrueIfFileIsDownloade
             string filePath = "TestFilePath";
 
             A.CallTo(() => fakefileShareService.DownloadReadMeFile(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored, A<string>.Ignored)).Returns(isFileDownloaded);
-            isFileDownloaded = await fulfilmentFileShareService.DownloadReadMeFile(filePath, batchId, exchangeSetRootPath,null);
-           
+            isFileDownloaded = await fulfilmentFileShareService.DownloadReadMeFile(filePath, batchId, exchangeSetRootPath, null);
+
             Assert.AreEqual(true, isFileDownloaded);
         }
 
         [Test]
         public async Task WhenRequestDownloadReadMeFile_ThenReturnsFalseIfFileIsNotDownloaded()
         {
             bool isFileDownloaded = false;
-            string batchId = "7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272";
-            string exchangeSetRootPath = @"D:\\Downloads";
+            string batchId = "7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272";            
             string filePath = "TestFilePath";
-
             A.CallTo(() => fakefileShareService.DownloadReadMeFile(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored, A<string>.Ignored)).Returns(isFileDownloaded);
-            isFileDownloaded = await fulfilmentFileShareService.DownloadReadMeFile(filePath, batchId, exchangeSetRootPath,null);
-
+            isFileDownloaded = await fulfilmentFileShareService.DownloadReadMeFile(filePath, batchId, fakeExchangeSetRootPath, null);
             Assert.AreEqual(false, isFileDownloaded);
         }
+
+        [Test]
+        public async Task WhenValidUploadZipFileRequest_ThenReturnTrue()
+        {
+            fakeIsFileUploaded = true;                   
+            A.CallTo(() => fakefileShareService.UploadZipFileForExchangeSetToFileShareService(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored)).Returns(fakeIsFileUploaded);
+            fakeIsFileUploaded = await fulfilmentFileShareService.UploadZipFileForExchangeSetToFileShareService(fakeBatchId, fakeExchangeSetRootPath, null);
+            Assert.AreEqual(true, fakeIsFileUploaded);
+        }
+
+        [Test]
+        public async Task WhenInvalidUploadZipFileRequest_ThenReturnFalse()
+        {
+            fakeIsFileUploaded = false;                       
+            A.CallTo(() => fakefileShareService.UploadZipFileForExchangeSetToFileShareService(A<string>.Ignored, A<string>.Ignored, A<string>.Ignored)).Returns(fakeIsFileUploaded);
+            fakeIsFileUploaded = await fulfilmentFileShareService.UploadZipFileForExchangeSetToFileShareService(fakeBatchId, string.Empty, null);
+            Assert.AreEqual(false, fakeIsFileUploaded);
+        }
```

Comment:
What are we trying to achieve with these two tests?

Are these two tests testing the return value as that is determined by the fake or should we be testing that this method calls the "UploadZipFileForExchangeSetToFileShareService" method on the fileShareService object?

i.e. the only responsibility that this method has is to call the method of the same name on the fileShareService object and it has no direct responsibility for the value returned.

Perhaps this should be a single test with a different name?

Does this make sense?

Reply:
In one test case, we are passing fake path for parameter ExchangeSetRootPath, where test case will pass and return true. 
and in other test case we are passing empty string for parameter ExchangeSetRootPath, where test case will fail and return false. Hence we have written two test cases.